var/spack/repos/builtin.mock/packages/garply/package.py

```diff
@@ -96,13 +98,28 @@ class Garply
             '-O2', '-g', '-DNDEBUG', '-fPIC',
             '-o', 'garplinator.cc.o',
             '-c', '%s/garply/garplinator.cc' % self.stage.source_path)
-        gpp('-fPIC', '-O2', '-g', '-DNDEBUG', '-shared',
-            '-Wl,-soname,libgarply.so', '-o', 'libgarply.so', 'garply.cc.o')
-        gpp('-O2', '-g', '-DNDEBUG', '-rdynamic',
-            'garplinator.cc.o', '-o', 'garplinator',
-            '-Wl,-rpath,%s' % prefix.lib64,
-            'libgarply.so')
-        copy('libgarply.so', '%s/libgarply.so' % prefix.lib64)
+        if sys.platform == 'darwin':
+            gpp('-fPIC', '-O2', '-g', '-DNDEBUG', '-dynamiclib',
+                '-Wl,-headerpad_max_install_names', '-o', 'libgarply.dylib',
+                '-install_name', '@rpath/libgarply.dylib',
+                'garply.cc.o')
+            gpp('-O2', '-g', '-DNDEBUG', '-Wl,-search_paths_first',
+                '-Wl,-headerpad_max_install_names',
+                'garplinator.cc.o', '-o', 'garplinator',
+                '-Wl,-rpath,%s' % prefix.lib64,
+                'libgarply.dylib')
+            mkdirp(prefix.lib64)
+            copy('libgarply.dylib', '%s/libgarply.dylib' % prefix.lib64)
+        else:
+            gpp('-fPIC', '-O2', '-g', '-DNDEBUG', '-shared',
+                '-Wl,-soname,libgarply.so',
+                '-o', 'libgarply.so', 'garply.cc.o')
+            gpp('-O2', '-g', '-DNDEBUG', '-rdynamic',
+                'garplinator.cc.o', '-o', 'garplinator',
+                '-Wl,-rpath,%s' % prefix.lib64,
+                'libgarply.so')
+            mkdirp(prefix.lib64)
+            copy('libgarply.so', '%s/libgarply.so' % prefix.lib64)
```

Comment:
Not necessary, but would it be clearer to put all this in a makefile in the garply directory?  I think it might make it more readable.

Reply:
Same for Corge, etc.

Reply:
The original package uses cmake. I just captured the commands used.